lib/openhab/log/logger.rb

```diff
@@ -60,8 +72,13 @@ def initialize(name)
       def clean_backtrace(error)
         return error if debug_enabled?
 
-        backtrace = error.backtrace_locations.reject { |line| INTERNAL_CALL_REGEX.match? line.to_s }
-        error.set_backtrace(backtrace.map(&:to_s))
+        if error.respond_to? :backtrace_locations
+          backtrace = error.backtrace_locations.reject { |line| INTERNAL_CALL_REGEX.match? line.to_s }
+          error.set_backtrace(backtrace.map(&:to_s))
```

Comment:
```suggestion
          backtrace = error.backtrace_locations.map(&:to_s).reject { |line| INTERNAL_CALL_REGEX.match? line }
          error.set_backtrace(backtrace)
```

Reply:
The `backtrace_locations` method doesn't return strings, but an Array of `Thread::Backtrace::Location`. However, the `set_backtrace` method requires an array of strings. See https://ruby-doc.org/core-2.5.8/Exception.html. Don't really understand the reason for it, but that's why it looks like it does.

Reply:
Yes, I'm just trying to avoid calling to_s twice here

Reply:
Ah, missed that you inserted `map(&:to_s)` before the reject, that would work!